### ISSUE TITLE
UI - Electron app  - Not able to execute scripts successfully when the tools are built using electron app.

### ISSUE THREAD

**Issue by sangee2004** [gptscript-ai/gptscript#687]
gptscript version v0.0.0-dev-7ff3fa1f-dirty

**Steps to reproduce the problem:**
1. Make sure your gptscript cache directory is empty - (`~/Library/Caches/gptscript`)
2. Lauch electron app
3. Execute the following gptscript in electron app:
```
Name: test1
Context: github.com/gptscript-ai/clio/context
Chat: true

You are a good assistant. Wait for user to ask you a question.
```

Following error shows up when script is loaded

<img width="1012" alt="Screenshot 2024-07-31 at 4 27 06 PM" src="https://github.com/user-attachments/assets/8203c93a-8e3e-4473-8a15-d7754838cd37">

4. Execute this script 
```
Tools: github.com/gptscript-ai/answers-from-the-internet, github.com/gptscript-ai/dalle-image-generation
Chat: true

You are a good assistant. Introduce yourself and wait for the user to ask you a question. Always include source of the search results in the response.
``` 
Following error is seen when `answers-from-the-internet` is getting run

<img width="975" alt="Screenshot 2024-07-31 at 4 28 11 PM" src="https://github.com/user-attachments/assets/8257282e-06b8-4023-b1aa-39eefef62480">

The same errors will also be seen when executing these scripts using `ui` or `tui`

```
%gptscript chat_internet.gpt 

  Hello! I'm here to assist you with any questions or tasks you have. How can I help you today?                                     

> who won 2024 superbowl?

    ┌────────────────────────────────────────────────────────────────────┐
    │ Call Arguments:                                                    │
    │                                                                    │
    │ answersFromTheInternet {"question":"Who won the 2024 Super Bowl?"} │
    └────────────────────────────────────────────────────────────────────┘
                                                                          
  Running answers-from-the-internet from github.com/gptscript-ai/answers-from-the-internet                                          

    ┌──────────────────────────────────────────────────────────────────────────────────┐
    │ ERROR: got (exit status 1) while running tool, OUTPUT:                           │
    │ > tool                                                                           │
    │ > node --no-warnings --loader ts-node/esm src/server.ts                          │
    │                                                                                  │
    │                                                                                  │
    │ node:internal/process/esm_loader:34                                              │
    │       internalBinding('errors').triggerUncaughtException(                        │
    │                                 ^                                                │
    │ [Object: null prototype] {                                                       │
    │   [Symbol(nodejs.util.inspect.custom)]: [Function: [nodejs.util.inspect.custom]] │
    │ }                                                                                │
    │                                                                                  │
    │ Node.js v21.7.0                                                                  │
    └──────────────────────────────────────────────────────────────────────────────────┘
                                                                                        
  It seems there was an issue retrieving the information. Let me try again.                                                         


    ┌────────────────────────────────────────────────────────────────────┐
    │ Call Arguments:                                                    │
    │                                                                    │
    │ answersFromTheInternet {"question":"Who won the 2024 Super Bowl?"} │
    └────────────────────────────────────────────────────────────────────┘
                                                                          
  Running answers-from-the-internet from github.com/gptscript-ai/answers-from-the-internet                                          

    ┌──────────────────────────────────────────────────────────────────────────────────┐
    │ ERROR: got (exit status 1) while running tool, OUTPUT:                           │
    │ > tool                                                                           │
    │ > node --no-warnings --loader ts-node/esm src/server.ts                          │
    │                                                                                  │
    │                                                                                  │
    │ node:internal/process/esm_loader:34                                              │
    │       internalBinding('errors').triggerUncaughtException(                        │
    │                                 ^                                                │
    │ [Object: null prototype] {                                                       │
    │   [Symbol(nodejs.util.inspect.custom)]: [Function: [nodejs.util.inspect.custom]] │
    │ }                                                                                │
    │                                                                                  │
    │ Node.js v21.7.0                                                                  │
    └──────────────────────────────────────────────────────────────────────────────────┘
                                                                                        
  I'm currently unable to retrieve the information about the 2024 Super Bowl winner. You might want to check the latest news or     
  sports websites for the most up-to-date information. If you have any other questions or need assistance with something else, feel 
  free to ask!                                                                                                                      

 % gptscript mytest1.gpt       
2024/07/31 16:31:19 failed resolving github.com/gptscript-ai/clio/context from ./mytest1.gpt: failed resolving github.com/gptscript-ai/context/multi-agent@adeedfa217075788d0bab38b505e8f4cb7d57333 from https://raw.githubusercontent.com/gptscript-ai/clio/b40b39f8c452abc8bcf9bb80f3c460f4f1821642/context/tool.gpt: failed resolving ../at-syntax from https://raw.githubusercontent.com/gptscript-ai/context/adeedfa217075788d0bab38b505e8f4cb7d57333/multi-agent/tool.gpt: failed resolving ./atsyntax.gpt from https://raw.githubusercontent.com/gptscript-ai/context/adeedfa217075788d0bab38b505e8f4cb7d57333/at-syntax: error loading https://raw.githubusercontent.com/gptscript-ai/context/adeedfa217075788d0bab38b505e8f4cb7d57333/atsyntax.gpt: error loading https://raw.githubusercontent.com/gptscript-ai/context/adeedfa217075788d0bab38b505e8f4cb7d57333/atsyntax.gpt: 404 Not Found

```

**Note** - The only way to get around the problem for executing these scripts in `TUI` is by using `--disable-cache` option.

For getting around the problem seen with launching apps in `--ui` or electron app , gptscript cache dir needs to be removed completely.   Then make sure to launch the scripts having tools in `ui` or `tui` first (not electron app).

**Comment by tylerslaton** [comment]
@sangee2004 Are you able to replicate this now that you cannot run arbitrary scripts?

**Comment by sangee2004** [comment]
Able to work successfully with  github.com/gptscript-ai/answers-from-the-internet and github.com/gptscript-ai/dalle-image-generation tools when it gets built for the first time using electron app.

Closing this issue.
